### PR TITLE
feat(ui-react-native): Add theming for PasswordField primitive

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

Note: The `styles` were not getting applied in `PasswordField.tsx`. This PR also fixes that issue. 

#### Description of how you validated changes

Wrote unit tests, ran in iOS and Android. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ioanabrooks — APPROVED** [review]

File: packages/react-native/src/primitives/PasswordField/PasswordField.tsx [hunk 1]
```diff
@@ -24,12 +31,13 @@ export default function PasswordField({
       {...rest}
       disabled={disabled}
       secureTextEntry={obscureText}
+      style={[themedStyle.container, style]}
```
Comment (calebpollman) [inline packages/react-native/src/primitives/PasswordField/PasswordField.tsx:34]: Is this a `container` or the field itself?

Comment (joebuono) [reply packages/react-native/src/primitives/PasswordField/PasswordField.tsx:34]: Currently, `TextField` applies the `style` prop to the container, not the field itself. ([link](https://github.com/aws-amplify/amplify-ui/blob/rna/main/packages/react-native/src/primitives/TextField/TextField.tsx#L39)). Perhaps we should have a separate `field` style for `PasswordField`?

Comment (joebuono) [reply packages/react-native/src/primitives/PasswordField/PasswordField.tsx:34]: Yes I think so ([link](https://github.com/aws-amplify/amplify-ui/blob/rna/main/packages/react-native/src/primitives/TextField/TextField.tsx#L45))

Comment (calebpollman) [reply packages/react-native/src/primitives/PasswordField/PasswordField.tsx:34]: Got it. Definitely not within the scope of this PR but we may want to rethink that

**Review by calebpollman — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**